dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestEmailSenderTest.kt

```diff
@@ -0,0 +1,261 @@
+            senderEmail,
+            senderName,
+            semicolonSeparatedInternalReceiverEmails,
+            semicolonSeparatedInternalCcEmails,
+            mockCompanyGetter,
+        )
+        mockEmailSender = mock(EmailSender::class.java)
+        singleDataRequestEmailSender = SingleDataRequestEmailSender(
+            mockEmailSender,
+            singleDataRequestEmailBuilder,
+            singleDataRequestInternalEmailBuilder,
+        )
+        val mockCompanyInformation = mock(CompanyInformation::class.java)
+        `when`(mockCompanyInformation.companyName).thenReturn(companyName)
+        `when`(mockCompanyGetter.getCompanyInfo(companyIdentifier)).thenReturn(mockCompanyInformation)
+    }
+
+    @Test
+    fun `validate that no email is sent if there are no reporting periods provided`() {
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
+                dataType,
+                listOfReportingPeriods = listOf(),
+                contactList = listOf("receiver@abc.de", "otherreceiver@something.else"),
+                message = defaultMessage,
+            ),
+            DataRequestCompanyIdentifierType.Isin,
+            companyIdentifier,
+        )
+        assertNumEmailsSentEquals(0)
+    }
+
+    @Test
+    fun `validate that an internal email is sent if there is no Dataland company ID provided`() {
+        expectSentEmailsToMatchInternalEmail(
+            DataRequestCompanyIdentifierType.Isin,
+        )
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
+                dataType,
+                listOfReportingPeriods = reportingPeriods,
+                contactList = listOf("receiver@abc.de", "otherreceiver@something.else"),
+                message = defaultMessage,
+            ),
+            DataRequestCompanyIdentifierType.Isin,
+            companyIdentifier,
+        )
+        assertNumEmailsSentEquals(1)
+    }
+
+    @Test
+    fun `validate that an internal email is sent if there are no contact provided`() {
+        expectSentEmailsToMatchInternalEmail(
+            DataRequestCompanyIdentifierType.DatalandCompanyId,
+            true,
+        )
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
+                dataType,
+                listOfReportingPeriods = reportingPeriods,
+                contactList = listOf(),
+                message = defaultMessage,
+            ),
+            DataRequestCompanyIdentifierType.DatalandCompanyId,
+            companyIdentifier,
+        )
+        assertNumEmailsSentEquals(1)
+    }
+
+    @Test
+    fun `validate that an external email is sent to the provided contact for a Dataland company ID`() =
```

Comment:
not a typo, but correct text would be ".. is sent to the provided contact**s** for a ...